src/Stratis.Bitcoin/Connection/PeerBanning.cs

```diff
@@ -126,5 +132,23 @@ public bool IsBanned(IPEndPoint endpoint)
 
             return peerAddress.BanUntil > this.dateTimeProvider.GetUtcNow();
         }
+
+        /// <inheritdoc />
+        public void UnBanPeer(IPEndPoint endpoint)
+        {
+            // Find all peers from the same IP and un-ban them.
+            List<PeerAddress> peerAddresses = this.peerAddressManager.FindPeersByIp(endpoint);
+            if (peerAddresses.Count == 0)
+            {
+                this.logger.LogTrace("(-)[NO_PEERS_TO_UNBAN]");
+                return;
+            }
+
+            foreach (var peerAddress in peerAddresses)
+            {
+                peerAddress.UnBan();
```

Comment:
Will this also update the store?

Reply:
Yes it does, this is tested in a set of new integration tests.